src/generation.rs

```diff
@@ -3511,4 +3541,142 @@ fn add_struct_derives<T: DataType>(data_type: &mut T, used_in_key: bool, is_enum
             }
         }
     }
+}
+
+fn generate_int(gen_scope: &mut GenerationScope, types: &IntermediateTypes) {
+    let ident = RustIdent::new(CDDLIdent::new("int"));
+    // todo: wasm bindings + to/from string traits
+    if CLI_ARGS.wasm {
+        let mut wrapper = create_base_wasm_wrapper(gen_scope, &ident, true);
+        let mut wasm_new = codegen::Function::new("new");
+        let mut new_if = codegen::Block::new("if x >= 0");
+        new_if.line("Self(core::Int::Uint(x as u64))");
+        let mut new_else = codegen::Block::new("else");
+        new_else.line("Self(core::Int::Nint((x + 1).abs() as u64))");
```

Comment:
we have +1 in a few places. Are these also all the cbor_event issue?

Reply:
Probably not? The cbor_event issue is just that `i64::MIN` isn't supported (it panics). There are other +1 and -1s with handling `nint` in CBOR as those start at -1 instead of 0 like with `uint`. Even just as an API restriction in cbor_event we can't get half the range of `nint` in general. With `preserve-encodings=true` we don't have any of these issues. It's possible we could use that API for all negative integers in the `preserve-encodings=false` with some extra work to calculate the encoding explicitly.